src/Frontend/Core/Engine/Page.php

```diff
@@ -372,7 +372,14 @@ private function addAlternateLinkForLanguage(string $language): void
             return;
         }
 
-        $url = Navigation::getUrl($this->pageId, $language);
+        // Get page data
+        $pageInfo = Model::getPage($this->pageId);
+
+        // Check if hreflang is set for language
+        if (!isset($pageInfo['data']['hreflang_' . $language])) {
+            return;
```

Comment:
if it isn't set I would check to see if there is a page with the same page id, to keep the old functionallity working, because that is how it is working for the home page etc at the moment, and I don't think that is a bad fallback. It would also bread backwards compatibity by changing this